File: backend/internal/domain/service/logger.go
package service

// Logger はロギング機能を提供するインターフェースです
type Logger interface {
	// Info は情報レベルのログを出力します
	Info(msg string, fields ...LogField)

	// Warn は警告レベルのログを出力します
	Warn(msg string, fields ...LogField)

	// Error はエラーレベルのログを出力します
	Error(msg string, fields ...LogField)

	// Debug はデバッグレベルのログを出力します
	Debug(msg string, fields ...LogField)
}

// LogField はログフィールドを表す構造体です
type LogField struct {
	Key   string
	Value interface{}
}

File: frontend/src/lib/logger.ts
class Logger {
  private readonly isDevelopment = import.meta.env.DEV;

  public debug(message: string, ...args: unknown[]) {
    if (this.isDevelopment) {
      console.log(`[DEBUG] ${message}`, ...args);
    }
  }

  public info(message: string, ...args: unknown[]) {
    if (this.isDevelopment) {
      console.info(`[INFO] ${message}`, ...args);
    }
  }

  public warn(message: string, ...args: unknown[]) {
    console.warn(`[WARN] ${message}`, ...args);
  }

  public error(message: string, ...args: unknown[]) {
    console.error(`[ERROR] ${message}`, ...args);
  }
}

export const logger = new Logger();
